### DISCUSSION TITLE
Être notifiés des erreurs applicatives dans notre outil de communication d'équipe

### DISCUSSION THREAD

**Discussion by NicolasRitouet** [betagouv/cap#6] [Besoin]
Solution possible : lier les erreurs Sentry à notre Mattermost.

Afin de pouvoir envoyer des notifications de sentry sur mattermost, il est nécessaire d'installer un plugin sur sentry:
https://github.com/NDrive/sentry-mattermost
Ca permettrait aux startups d'etre plus réactives, de pouvoir mieux communiquer sur les erreurs et de rendre le travail des devs plus visibles.

**Comment by revolunet** [comment]
J'ai mis en place un bridge sur scalingo, c'est à dispo si tu veux tester @NicolasRitouet, je peux te fournir les urls en privé.

<img width="423" alt="Capture d’écran 2024-02-09 à 16 52 12" src="https://github.com/betagouv/cap/assets/124937/3bb6a8af-44ce-4540-a724-fe45b6c869b6">

Configuration dans Sentry : 
 - In the PROJECT settings (`/settings/[org]/projects/[project]]/alerts/`), add the webhook URL, and add `?channel=[project]-alerts` if you want to specify a specific channel.
 - In the PROJECT alerts (`/organizations/[org]/alerts/rules/[project]`), add a Rule to send events to the "integration" "Webhooks".


[le code](https://github.com/betagouv/sentry-mattermost)

**Reply by mlbiche** [reply]
Salut @revolunet,
J'ai fait le test pour ma SE, l'alerte a bien été _triggerred_ dans Sentry, mais rien dans Mattermost. Pourtant j'ai bien affecté la valeur `https://chat.incubateur.anct.gouv.fr/hooks/[idDeMonHook]?channel=startup-monsuivisocial-monitoring` dans la configuration de Sentry.

Le test depuis Sentry (bouton `Test plugin`) donne :

```
There was an internal error with the Plugin, {\"id\":\"web.incoming_webhook.text.app_error\",\"message\":\"Aucun texte indiqué\",\"detailed_error\":\"\",\"request_id\":\"t1msfe9jmi883fk9g8tkbwxs8h\",\"status_code\":400}
```

Peut-être est-ce dû à la configuration du webhook Mattermost ?

**Reply by sneko** [reply]
@mlbiche à confirmer par @revolunet mais il me semble que ça a seulement été mis sur le Mattermost principal de beta.gouv puisqu'il faut une clef secrète pour poster les messages.

Est-il envisageable d'avoir votre canal de monitoring sur celui-ci ?

**Reply by mlbiche** [reply]
Si j'en crois le [code du _bridge_](https://github.com/betagouv/sentry-mattermost/blob/main/bridge.mjs#L69), seuls les canaux `*-sentry` sont autorisés. Donc normal que ça ne passe pas.

**EDIT :** ... mais ça ne semble effectivement pas être ça.

**Reply by mlbiche** [reply]
Je pense que si on a les alertes Sentry sur le Mattermost de beta.gouv, ça perdra son intérêt pour nous. On a déjà nos alertes Grafana sur le Mattermost de l'Incubateur, donc ça multiple les canaux à surveiller... À creuser, peut-être avec @NicolasRitouet. Ce n'est pas forcément pressé de notre côté.

**Reply by revolunet** [reply]
Oui le bridge est branché sur un webhook du mattermost beta.

L'url "secrete" à utiliser est indiquée ici : https://github.com/betagouv/beta-communaute-support/issues/44#issuecomment-1936237328

Le canal doit bien se terminer par `-sentry` actuellement. (on peut améliorer si besoin)

**Reply by mlbiche** [reply]
Hum... 🤔 Du coup peut-être qu'au sein de notre Incubateur on aurait intérêt à _forker_ votre bridge pour avoir une instance pour le Mattermost de l'Incub. Je préfère laisser @NicolasRitouet se prononcer sur ce sujet.

**Reply by revolunet** [reply]
ce serait plus simple oui; sinon on peut aussi améliorer le bridge pour qu'il puisse gérer plusieurs webhooks MM

**Reply by NicolasRitouet** [reply]
Désolé, je n'ai pas du tout eu le temps de me pencher sur le sujet. Nous sommes en train de planifier une migration de notre mattermost. Je m'en occuperai une fois la migration finie.
